pattern/factory-pattern.test.js

```
@@ -0,0 +1,137 @@
+// factory-pattern.test.js
+const fs = require('fs');
+const path = require('path');
+
+// Import the factory pattern code
+// Note: In a real project, you would use proper imports
+// This approach is used because we're testing a standalone file
+const filePath = path.join(__dirname, 'factory-pattern.js');
+eval(fs.readFileSync(filePath, 'utf8'));
```

Biome (1.9.4)

[error] eval() exposes to security risks and performance issues.

See the MDN web docs for more details.
Refactor the code so that it doesn't need to call eval().

(lint/security/noGlobalEval)